cheri-c-programming.tex

```diff
@@ -1174,6 +1176,54 @@ \section{TODO: CHERI compiler warnings and errors}
   Abstract and Introduction to mention this, and xref from potential causes
   above -- e.g., as relates to integer/pointer types, etc.}
 
+The CHERI Clang compiler includes many diagnostic warnings to identify code that is incompatible with \purecapCOrCpp{} or may result in behavioral differences.
+In many cases, a successful compilation that does not emit any CHERI-specific warnings will result in a functional spatially-safe program.
+However, some incompatibilities (e.g.\ memory allocators returning insufficiently aligned pointers) cannot yet be diagnosed statically.
+This section
+
+\arnote{TODO: Sensible formatting}
+\lstnewenvironment{compilerwarning}[1][]{\lstset{language={},breaklines=true,basicstyle=\small\ttfamily\bfseries,#1}}{}
+\begin{compilerwarning}
+cast from provenance-free integer type to pointer type will give pointer that can not be dereferenced
+\end{compilerwarning}
```

Comment:
A question I don't have a good answer to is: should we consider providing actual compiler output or is that too noisy?

Reply:
Do you mean provide something like this?
```
<source>:2:12: warning: cast from provenance-free integer type to pointer type will give pointer that can not be dereferenced [-Wcheri-capability-misuse]
    return (void*)l;
           ^
<source>:2:12: note: insert cast to intptr_t to silence this warning
1 warning generated.
```

I though using the compiler message as the heading should be enough, but I can definitely add more text.

Reply:
I could change the section heading to be `-Wcheri-capability-misuse`?

Reply:
Although actually I believe that is the warning group and not this specific warning...